topdown/eval.go

```diff
@@ -3307,6 +3383,7 @@ func (e evalEvery) eval(iter unifyIterator) error {
 	}
 
 	domain := e.e.closure(e.generator)
+	domain.suppressEarlyExit = true
```

Comment:
Need to investigate further if there are more places that need this.

Reply:
I _think_ other closures could be set/object/array comprehensions? 🤔